prepare for digiline_routing removal

BuckarooBanzay opened the pull request:
just in case
(plus revert of: https://github.com/pandorabox-io/pandorabox-mods/pull/688)

BuckarooBanzay committed:
prepare for digiline_routing removal

S-S-X commented:
There's now one suggestion for fixing components at https://github.com/pandorabox-io/pandorabox.io/issues/516

Would you like to review too, can timers become problem when used like that?
Should there be cleanup to prevent possible timer stacking?

I've also updated https://github.com/numberZero/digiline_routing/issues/9 a bit but I fear that changes in PR's might be too much for numberZero to handle fast enough.

For that of course repo can be forked (I'd prefer mt-mods) and then PR's can sit there for some time if needed to.

Probably it is already clear enough that digiline_routing mod is not really production ready but it still provides very good base for "hardware" digiline components, if few people are willing to put some time into that I think problems can be fixed.

Biggest problems with diode should be fairly easy fix, if that can be added maybe **approach could be disabling other components until problems are fixed** (same problems as with diode just bit more complex) .

BuckarooBanzay commented:
> There's now one suggestion for fixing components at pandorabox-io/pandorabox.io#516

I've switched to @Lejo1's fork for the time being (until/if NumberZero merges the changes)

> Would you like to review too, can timers become problem when used like that?
> Should there be cleanup to prevent possible timer stacking?

The async approach does sound good, but i wouldn't worry that much about recursion, lua has tail call optimization in _some_ cases (https://stackoverflow.com/questions/13303347/tail-call-optimization-in-lua)

> I've also updated numberZero/digiline_routing#9 a bit but I fear that changes in PR's might be too > much for numberZero to handle fast enough.

> For that of course repo can be forked (I'd prefer mt-mods) and then PR's can sit there for some time if needed to.

We can move that to the `mt-mods` repo if there are more changes/issues :+1: 

> Probably it is already clear enough that digiline_routing mod is not really production ready but it still provides very good base for "hardware" digiline components, if few people are willing to put some time into that I think problems can be fixed.

> Biggest problems with diode should be fairly easy fix, if that can be added maybe approach could be disabling other components until problems are fixed (same problems as with diode just bit more complex) .

How about disabling the the recipes and giving out a few stacks to responsible players?

**EDIT**: also: closing this, looks like the removal is not needed...
